libgnucash/backend/sql/gnc-sql-column-table-entry.cpp

```diff
@@ -152,6 +152,7 @@ GncSqlColumnTableEntryImpl<CT_STRING>::add_to_query(QofIdTypeConst obj_name,
         stream << s;
         vec.emplace_back (std::make_pair (std::string{m_col_name},
                                           quote_string(stream.str())));
+        g_free (s);
```

Comment:
If I understand the code right, there could be a problem here.  `get_row_value_from_object` can produce its result through several paths with mixed pointer ownership.  The `g_object_get` path returns a string owned by the caller, but the `qof_class_get_parameter_getter` path returns from `g_hash_table_lookup` and I think that's a direct copy of the pointer location into the hash table.  This string would need to be duplicated in order to have consistent memory ownership.

Reply:
Yeah I can't understand this code; g_object_get would occasionally return a newly malloc'd object which should be freed.

Reply:
So we need specializations of [`get_row_value_from_object<char*>(QofIdTypeConst obj_name,
                                                  const void* pObject, Z)`](https://github.com/Gnucash/gnucash/blob/906d4113c15267175b5ef40c38f9e1b9197263b5/libgnucash/backend/sql/gnc-sql-column-table-entry.hpp#L335) where `Z` is `std::true_type` and `std::false_type` that `g_strdup`s the rv from the Qof Getter. I'll do that when I have time this weekend.